common/djangoapps/student/views.py

```diff
@@ -583,20 +597,20 @@ def dashboard(request):
     # there is no verification messaging to display.
     verify_status_by_course = check_verify_status_by_course(
         user,
-        course_enrollment_pairs,
+        course_enrollments,
         all_course_modes
     )
     cert_statuses = {
-        course.id: cert_info(request.user, course, _enrollment.mode)
-        for course, _enrollment in course_enrollment_pairs
+        enrollment.course_id: cert_info(request.user, enrollment.course_overview, enrollment.mode)
+        for enrollment in course_enrollments
     }
 
     # only show email settings for Mongo course and when bulk email is turned on
     show_email_settings_for = frozenset(
-        course.id for course, _enrollment in course_enrollment_pairs if (
+        enrollment.course_id for enrollment in course_enrollments if (
             settings.FEATURES['ENABLE_INSTRUCTOR_EMAIL'] and
-            modulestore().get_modulestore_type(course.id) != ModuleStoreEnum.Type.xml and
-            CourseAuthorization.instructor_email_enabled(course.id)
+            modulestore().get_modulestore_type(enrollment.course_id) != ModuleStoreEnum.Type.xml and
```

Comment:
Correct - it's cached after the first access. It should not access Mongo again to determine the modulestore access after it has been cached.